pkg/codegen/go/gen_program_expression_test.go

```diff
@@ -0,0 +1,37 @@
+package gen
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/model"
+	"github.com/stretchr/testify/assert"
+)
+
+type exprTestCase struct {
+	hcl2Expr string
+	goCode   string
+}
+
+func TestLiteralExpression(t *testing.T) {
+	cases := []exprTestCase{
+		{hcl2Expr: "false", goCode: "false"},
+		{hcl2Expr: "true", goCode: "true"},
+		{hcl2Expr: "0", goCode: "0"},
+		{hcl2Expr: "3.14", goCode: "3.14"},
+		{hcl2Expr: "\"foo\"", goCode: `"foo"`},
```

Comment:
Consider adding tests for strings with quotes and backslashes.

Reply:
Will work on more robust testing for strings when I do templates (register allocation :))